packages/@aws-cdk/aws-glue-alpha/lib/jobs/job.ts

```diff
@@ -0,0 +1,559 @@
+    return this.onStateChange(id, JobState.TIMEOUT, options);
+  }
+
+  /**
+   * Create a CloudWatch metric.
+   *
+   * @param metricName name of the metric typically prefixed with `glue.driver.`, `glue.<executorId>.` or `glue.ALL.`.
+   * @param type the metric type.
+   * @param props metric options.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitoring-awsglue-with-cloudwatch-metrics.html
+   */
+  public metric(metricName: string, type: MetricType, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return new cloudwatch.Metric({
+      metricName,
+      namespace: 'Glue',
+      dimensionsMap: {
+        JobName: this.jobName,
+        JobRunId: 'ALL',
+        Type: type,
+      },
+      ...props,
+    }).attachTo(this);
+  }
+
+  /**
+   * Return a CloudWatch Metric indicating job success.
+   *
+   * This metric is based on the Rule returned by no-args onSuccess() call.
+   */
+  public metricSuccess(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return metricRule(this.metricJobStateRule('SuccessMetricRule', JobState.SUCCEEDED), props);
+  }
+
+  /**
+   * Return a CloudWatch Metric indicating job failure.
+   *
+   * This metric is based on the Rule returned by no-args onFailure() call.
+   */
+  public metricFailure(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return metricRule(this.metricJobStateRule('FailureMetricRule', JobState.FAILED), props);
+  }
+
+  /**
+   * Return a CloudWatch Metric indicating job timeout.
+   *
+   * This metric is based on the Rule returned by no-args onTimeout() call.
+   */
+  public metricTimeout(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return metricRule(this.metricJobStateRule('TimeoutMetricRule', JobState.TIMEOUT), props);
+  }
+
+  /**
+   * Creates or retrieves a singleton event rule for the input job state for use with the metric JobState methods.
+   *
+   * @param id construct id.
+   * @param jobState the job state.
+   * @private
+   */
+  private metricJobStateRule(id: string, jobState: JobState): events.Rule {
+    return this.node.tryFindChild(id) as events.Rule ?? this.onStateChange(id, jobState);
+  }
+
+  /**
+   * Returns the job arn
+   * @param scope
+   * @param jobName
+   */
+  protected buildJobArn(scope: constructs.Construct, jobName: string) : string {
+    return cdk.Stack.of(scope).formatArn({
+      service: 'glue',
+      resource: 'job',
+      resourceName: jobName,
+    });
+  }
+}
+
+/**
+ * A subset of Job attributes are required for importing an existing job
+ * into a CDK project. This is ionly used when using fromJobAttributes
+ * to identify and reference the existing job.
+ */
+export interface JobImportAttributes {
+  /**
+   * The name of the job.
+   */
+  readonly jobName: string;
+
+  /**
+   * The IAM role assumed by Glue to run this job.
+   *
+   * @default - undefined
+   */
+  readonly role?: iam.IRole;
```

Comment:
Are you asking if it should be optional or if it should be removed?  We do need a Role since every job requires an execution role, but if it's not provided we assign an UnknownPrincipal (line 468)